lib/chef/resource/lwrp_base.rb

```diff
@@ -58,7 +58,10 @@ def build_from_file(cookbook_name, filename, run_context)
           resource_class.class_from_file(filename)
 
           # Make a useful string for the class (rather than <Class:312894723894>)
-          resource_class.instance_eval do
+          resource_class.class_eval do
+            define_method(:kind_of?)      { |other| other.class <= resource_class }
```

Comment:
Lets add a comment that explains why we need this.  Something along the lines of...

If you have a subclass of the LWRP class (like we make to support the deprecated class lookup under `Chef::Resource`) and you call `subclass.kind_of? LwrpClass` that will return false... Unless! we override those methods to make sure compare `kind_of?` with the real resource class.

@jkeiser Is my understanding of this correct?